File: Online-Stock-Span.java
1class StockSpanner {
2
3    List<Integer> prices ;
4
5    public StockSpanner() {
6        prices = new ArrayList<>();
7    }
8    
9    public int next(int price) {
10        prices.add(price);
11        int res = 0 ; 
12        int i = prices.size() -1 ; 
13        while(i >= 0 && price >= prices.get(i)){
14            res++; 
15            i--; 
16        }
17        return res; 
18    }
19}
20
21/**
22 * Your StockSpanner object will be instantiated and called as such:
23 * StockSpanner obj = new StockSpanner();
24 * int param_1 = obj.next(price);
25 */

File: Online-Stock-Span.cpp
1class StockSpanner {
2stack<pair<int,int>> st;
3public:
4    StockSpanner() {
5      
6    }
7    
8    int next(int price) {
9        int cnt = 1 ;
10        while(!st.empty() && price >= st.top().first){
11            cnt+= st.top().second; 
12            st.pop();
13        }
14        st.push({price, cnt});
15
16        return st.top().second;
17    }
18};
19
20/**
21 * Your StockSpanner object will be instantiated and called as such:
22 * StockSpanner* obj = new StockSpanner();
23 * int param_1 = obj->next(price);
24 */